code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -20,6 +20,14 @@
 	var/enroute = 0
 	var/list/targets = list()
 	var/attacked_times = 0
+	var/list/target_type_validator_map = list()
+	var/lowest_health = INFINITY // Max you can get
+
+/mob/living/bot/secbot/Initialize()
+	. = ..()
+	target_type_validator_map.add(/mob/living = CALLBACK(src, ./proc/living), /other/path = TRUE)

```

Comment:
ooookay, that's not how you add stuff to a map. This should be:
```dm
	target_type_validator_map[/mob/living] = CALLBACK(src, ./proc/living)
	target_type_validator_map[/obj/mecha] = CALLBACK(src, ./proc/mecha)
	target_type_validator_map[/obj/machinery/bot] = CALLBACK(src, ./proc/bot)
```

Reply:
I was confused by your comment before.